File: LeetCode/Easy/0094-binary-tree-inorder-traversal/0094-binary-tree-inorder-traversal.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def inorderTraversal(self, root: Optional[TreeNode]) -> List[int]:
        ret = []
        
        def dfs(node):
            if not node:
                return
            
            dfs(node.left)
            ret.append(node.val)
            dfs(node.right)
        
        dfs(root)
        
        return ret

File: LeetCode/Easy/0094-binary-tree-inorder-traversal/0094-binary-tree-inorder-traversal.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func inorderTraversal(root *TreeNode) []int {
    ret := make([]int, 0)
    
    dfs := func(node *TreeNode){}
    dfs = func(node *TreeNode){
        if node == nil { return }
        
        dfs(node.Left)
        ret = append(ret, node.Val)
        dfs(node.Right)
    }
    
    dfs(root)
    
    return ret
}